js-apps/meep-frontend/src/js/containers/exec/view-pane.js

```diff
@@ -0,0 +1,382 @@
+    if (sandboxCfg && sandboxCfg[sandboxName] && sandboxCfg[sandboxName].dashboardView2) {
+      return sandboxCfg[sandboxName].dashboardView2;
+    } else {
+      return VIEW_NAME_NONE;
+    }
+  }
+
+  changeView1(viewName, sdbCfg) {
+    var sandboxCfg = updateObject({}, sdbCfg);
+    sandboxCfg[this.props.sandbox].dashboardView1 = viewName;
+    this.props.changeSandboxCfg(sandboxCfg);
+  }
+
+  changeView2(viewName) {
+    var sandboxCfg = updateObject({}, this.props.sandboxCfg);
+    sandboxCfg[this.props.sandbox].dashboardView2 = viewName;
+    this.props.changeSandboxCfg(sandboxCfg);
+  }
+
+  populateDashboardList(dashboardViewsList, dashboardOptions) {
+    for (var i = 0; i < dashboardOptions.length; i++) {
+      var dashboard = dashboardOptions[i];
+      if ((dashboard.label !== '') && (dashboard.value !== '') && (dashboard.value.indexOf(showInExecStr) !== -1)) {
+        dashboardViewsList.push(dashboard.label);
+      }
+    }
+  }
+  
+  onViewPaneClose(e) {
+    e.preventDefault();
+    this.props.changeView('');
+    this.props.changeViewStatus('');
+    this.props.onClose(e);
+  }
+
+  updateElement(values) {
+    if (values === null) {
+      this.props.resetSelectedScenarioElement();
+    } else {
+      var element = updateObject({}, this.props.selectedScenarioElement);
+      element = updateObject(element, values);
+      this.props.changeSelectedScenarioElement(element);
+    }
+  }
+
+  getRoot() {
+    return d3.hierarchy(this.props.displayedScenario, getScenarioNodeChildren);
+  }
+
+  render() {
+    if (this.props.page !== PAGE_EXECUTE || this.props.hide) {
+      return null;
+    }
+    
+    const view1Name = this.getView1();
+    const view2Name = this.getView2();
+
+    const root = this.getRoot();
+    const nodes = root.descendants();
+    const apps = nodes.filter(isApp);
+    const appIds = apps.map(a => a.data.name).sort();
+    const ues = nodes.filter(isUe);
+    const ueIds = ues.map(a => a.data.name).sort();
+    const poas = nodes.filter(isPoa);
+    const poaIds = poas.map(a => a.data.name).sort();
+    appIds.unshift('None');
+    ueIds.unshift('None');
+    poaIds.unshift('None');
+
+    // Populate Dashboard view list using links from monitoring tab
+    var dashboardViewsList = [
+      VIEW_NAME_NONE,
+      MAP_VIEW,
+      NET_TOPOLOGY_VIEW
+    ];
+    this.populateDashboardList(dashboardViewsList, DEFAULT_DASHBOARD_OPTIONS);
+    this.populateDashboardList(dashboardViewsList, this.props.dashboardOptions);
+
+    return (
+      <div style={{ padding: 10 }}>
+        <div style={styles.block}>
+          <Typography use="headline6">Dashboard View</Typography>
+        </div>
+        <ViewSelect
+          viewOptions={this.props.viewOptions}
+          onChange={event => {
+            this.props.changeView(event.target.value);
+            this.props.changeViewStatus('');
+          }}
+          value={this.props.currentView}
+        />
+        <ViewSelectFields
```

Comment:
Recommend merging view-specific-pane into this file. There is no need for a separate component in this specific case.

Reply:
actually agree with separating it or not... i am asking not to because it prevents to do few if statement to see which view we are on so that the proper values can be populated. I believe using a different component is useful for readability and if more views are added that would make it even worse for the if view number... but it was renamed to dashCfgDetailPane....

Reply:
I'm okay with leaving in a separate component.